sw/device/sca/kmac_serial.c

```diff
@@ -50,11 +50,13 @@ enum {
    * Number of cycles (at `kClockFreqCpuHz`) that Ibex should sleep to minimize
    * noise during SHA3 operations. Caution: This number should be chosen to
    * provide enough time. Otherwise, Ibex might wake up while SHA3 is still busy
-   * and disturb the capture. Currently, we use a start trigger delay of 40
-   * clock cycles and the scope captures 200 clock cycles at kClockFreqCpuHz
-   * (2000 samples).
+   * and disturb the capture. Currently, we use a start trigger delay of 320
+   * clock cycles and the scope captures 160 clock cycles at kClockFreqCpuHz
+   * (3200 samples). On the scope side, an offset of 395 clock cycles (7900
+   * samples) is used to ignore 1) the start trigger delay for the PROCESS
+   * command, and 2) the absorb phase for the fixed prefix.
    */
-  kIbexSha3SleepCycles = 800,
+  kIbexSha3SleepCycles = 1180,
```

Comment:
We are increasing `kIbexAesSleepCycles` by 280 but `kIbexSha3SleepCycles` by 380 although the change in delay is equal to 280 cycles. What is the reason for that difference?

Reply:
Well spotted @abdullahvarici !

For SHA3/KMAC, the delay is actually observed twice as there are two commands that Ibex sends to the KMAC peripheral. First a `START` to initialize the Keccak core and start loading data from the message FIFO into the Keccak state. Second, a `PROCESS` command to add the padding and kick off the actual Keccak processing. On FPGA, Ibex can send both commands back2back but the command delay is applied to both. So the increase in total delay is actually 2x 280 cycles. 

The total number of sleep cycles for KMAC I estimated as
- 2x the delay = 640 cycles
- Loading key, padding + SHA3 processing + loading the message afterwards= 160 cycles
- 380 spare cycles

Reply:
Thanks @vogelpi for the explanation!